module/zfs/zfs_ioctl.c

```diff
@@ -4605,6 +4623,29 @@ zfs_ioc_userspace_upgrade(zfs_cmd_t *zc)
 	return (error);
 }
 
+/*
+ * inputs:
+ * zc_name		name of filesystem
+ *
+ * outputs:
+ * none
+ */
+static int
+zfs_ioc_userobjspace_upgrade(zfs_cmd_t *zc)
+{
+	int err;
+	zfs_sb_t *zsb;
+
+	err = zfs_sb_hold(zc->zc_name, FTAG, &zsb, B_FALSE);
+	if (err != 0)
```

Comment:
hmm.. isn't this exactly what we want? If an objset is not mounted, we should mount it and upgrade. This may be rare for a regular ZFS objset but should be common for Lustre due to ''canmount=off' option.